File: gen_scripts/php/f00.loop.php
#!/usr/bin/env php
<?php
// testbox: requires=posix
// testbox: title="shell_exec (ls) with foreach collection"
// collection loop on results
foreach (preg_split("/\s+/", rtrim(shell_exec('ls dirtest'))) as $item) {
   if (is_dir("dirtest/$item")) {
       echo "$item is a directory\n";
   } else {
       echo "$item is not a directory\n";
   }
}
?>

File: gen_scripts/python2/f00.loop.py
#!/usr/bin/env python
import os

# collection loop on buffered output from subshell
for item in os.popen("ls dirtest").readlines():
  # test if item is directory
  path = "dirtest/" + item
  if os.path.isdir(path.rstrip()):
    print "%s is a directory" % item.rstrip()
  else:
    print "%s is not a directory" % item.rstrip()
